from PyQt5 import QtCore, QtGui, QtWidgets
import mysql.connector as db 

class Ui_bill(object):

    def Insert(self):
        id = self.id_in.text()
        name = self.name_in.text()
        price = int(self.price_in.text())
        qty = int(self.qty_in.text())
        total = str(price * qty)
        self.tot_in.setText(str(total))
        all = f"Id:{id}, Product Name:{name}, Product Proce:{price}, Quantity{qty}"
        self.all.setText(all)

        if(name=="" or price=="" or qty=="" ):
            QtWidgets.QMessageBox.information(None, "Alert", "Please Enter all Fields!")
        else:
            con = db.connect(host="localhost", user="root", password="", database="bitdb")
            cursor = con.cursor()
            query = "INSERT INTO bill values('"+id+"','"+name+"','"+str(price)+"', '"+str(qty)+"', '"+str(total)+"')"
            cursor.execute(query)
            cursor.execute("commit")

    def Update(self):
        Id = self.id_in.text()
        Name = self.name_in.text()
        Price = int(self.price_in.text())
        Qty = int(self.qty_in.text())
        Total = str(Price * Qty)

        if(Id=="" or Name=="" or Price=="" or Qty=="" or Total==""):
            QtWidgets.QMessageBox.warning("All fields are required..!")
        else:
                con = db.connect(host="localhost", user="root", password="", database="bitdb")
                cursor = con.cursor()
                cursor.execute("Update bill set name='"+Name+"', price='"+str(Price)+"', qty='"+str(Qty)+"', total='"+str(Total)+"'where id='"+Id+"'")
                con.commit()
                self.name_in.clear()
                self.price_in.clear()
                self.qty_in.clear()
                QtWidgets.QMessageBox.information(None, "Success", "Details Updated..!")


    def FetchData(self):
        if(self.id.text()==""):
            QtWidgets.QMessageBox.information(None, "Alert", "Please Enter all Fields!")
        else:
            con = db.connect(host="localhost", user="root", password="", database="bitdb")
            cursor = con.cursor()
            cursor.execute("select * from bill where id='"+self.id_in.text()+"'")
            res = cursor.fetchall()

            for row in res:
                self.name_in.setText(row[1])
                self.price_in.setText(str(row[2]))
                self.qty_in.setSpecialValueText(str(row[3]))
                self.total.setText(str(row[4]))

            cursor.close()
        con.close()

    def Delete(self):
        if(self.id.text()==""):
            QtWidgets.QMessageBox.warning("Id is required")
        else:
            con = db.connect(host="localhost", user="root", password="", database="bitdb")
            cursor = con.cursor()
            cursor.execute("delete from bill where id='"+self.id_in.text()+"'")
            con.commit()
            QtWidgets.QMessageBox.information(None, "Success", "Deleted Successfully..!")
            con.close()


    def setupUi(self, bill):
        bill.setObjectName("bill")
        bill.resize(798, 488)
        palette = QtGui.QPalette()
        brush = QtGui.QBrush(QtGui.QColor(255, 255, 255))
        brush.setStyle(QtCore.Qt.SolidPattern)
        palette.setBrush(QtGui.QPalette.Active, QtGui.QPalette.Base, brush)
        brush = QtGui.QBrush(QtGui.QColor(200, 0, 0))
        brush.setStyle(QtCore.Qt.SolidPattern)
        palette.setBrush(QtGui.QPalette.Active, QtGui.QPalette.Window, brush)
        brush = QtGui.QBrush(QtGui.QColor(255, 255, 255))
        brush.setStyle(QtCore.Qt.SolidPattern)
        palette.setBrush(QtGui.QPalette.Inactive, QtGui.QPalette.Base, brush)
        brush = QtGui.QBrush(QtGui.QColor(200, 0, 0))
        brush.setStyle(QtCore.Qt.SolidPattern)
        palette.setBrush(QtGui.QPalette.Inactive, QtGui.QPalette.Window, brush)
        brush = QtGui.QBrush(QtGui.QColor(200, 0, 0))
        brush.setStyle(QtCore.Qt.SolidPattern)
        palette.setBrush(QtGui.QPalette.Disabled, QtGui.QPalette.Base, brush)
        brush = QtGui.QBrush(QtGui.QColor(200, 0, 0))
        brush.setStyle(QtCore.Qt.SolidPattern)
        palette.setBrush(QtGui.QPalette.Disabled, QtGui.QPalette.Window, brush)
        bill.setPalette(palette)
        bill.setAutoFillBackground(True)
        self.label = QtWidgets.QLabel(bill)
        self.label.setGeometry(QtCore.QRect(310, 20, 191, 41))
        font = QtGui.QFont()
        font.setPointSize(15)
        font.setBold(True)
        font.setUnderline(True)
        font.setWeight(75)
        self.label.setFont(font)
        self.label.setObjectName("label")
        self.name = QtWidgets.QLabel(bill)
        self.name.setGeometry(QtCore.QRect(120, 170, 121, 20))
        font = QtGui.QFont()
        font.setPointSize(10)
        font.setBold(True)
        font.setWeight(75)
        self.name.setFont(font)
        self.name.setObjectName("name")
        self.price = QtWidgets.QLabel(bill)
        self.price.setGeometry(QtCore.QRect(120, 220, 55, 16))
        font = QtGui.QFont()
        font.setPointSize(10)
        font.setBold(True)
        font.setWeight(75)
        self.price.setFont(font)
        self.price.setObjectName("price")
        self.qty = QtWidgets.QLabel(bill)
        self.qty.setGeometry(QtCore.QRect(120, 270, 91, 16))
        font = QtGui.QFont()
        font.setPointSize(10)
        font.setBold(True)
        font.setWeight(75)
        self.qty.setFont(font)
        self.qty.setObjectName("qty")
        self.tot = QtWidgets.QLabel(bill)
        self.tot.setGeometry(QtCore.QRect(120, 320, 55, 16))
        font = QtGui.QFont()
        font.setPointSize(10)
        font.setBold(True)
        font.setWeight(75)
        self.tot.setFont(font)
        self.tot.setObjectName("tot")
        self.name_in = QtWidgets.QLineEdit(bill)
        self.name_in.setGeometry(QtCore.QRect(270, 170, 113, 22))
        self.name_in.setObjectName("name_in")
        self.price_in = QtWidgets.QLineEdit(bill)
        self.price_in.setGeometry(QtCore.QRect(270, 220, 113, 22))
        self.price_in.setObjectName("price_in")
        self.tot_in = QtWidgets.QLineEdit(bill)
        self.tot_in.setGeometry(QtCore.QRect(270, 320, 113, 22))
        self.tot_in.setObjectName("total")
        self.insert = QtWidgets.QPushButton(bill)
        self.insert.setGeometry(QtCore.QRect(590, 120, 93, 28))
        self.insert.setObjectName("insert")
        self.update = QtWidgets.QPushButton(bill)
        self.update.setGeometry(QtCore.QRect(590, 170, 93, 28))
        self.update.setObjectName("update")
        self.delete = QtWidgets.QPushButton(bill)
        self.delete.setGeometry(QtCore.QRect(590, 220, 93, 28))
        self.delete.setObjectName("delete")
        self.select = QtWidgets.QPushButton(bill)
        self.select.setGeometry(QtCore.QRect(590, 270, 93, 28))
        self.select.setObjectName("select")
        self.all = QtWidgets.QLineEdit(bill)
        self.all.setGeometry(QtCore.QRect(200, 410, 401, 31))
        self.all.setObjectName("all")
        self.id = QtWidgets.QLabel(bill)
        self.id.setGeometry(QtCore.QRect(120, 120, 55, 16))
        font = QtGui.QFont()
        font.setPointSize(10)
        font.setBold(True)
        font.setWeight(75)
        self.id.setFont(font)
        self.id.setObjectName("id")
        self.id_in = QtWidgets.QLineEdit(bill)
        self.id_in.setGeometry(QtCore.QRect(270, 120, 113, 22))
        self.id_in.setObjectName("id_in")
        self.qty_in = QtWidgets.QSpinBox(bill)
        self.qty_in.setGeometry(QtCore.QRect(270, 270, 111, 22))
        self.qty_in.setObjectName("qty_in")

        self.retranslateUi(bill)
        QtCore.QMetaObject.connectSlotsByName(bill)
        self.insert.clicked.connect(self.Insert)
        self.select.clicked.connect(self.FetchData)
        self.update.clicked.connect(self.Update)
        self.delete.clicked.connect(self.Delete)

    def retranslateUi(self, bill):
        _translate = QtCore.QCoreApplication.translate
        bill.setWindowTitle(_translate("bill", "Billing System"))
        self.label.setText(_translate("bill", "Billing System"))
        self.name.setText(_translate("bill", "Product Name"))
        self.price.setText(_translate("bill", "Price"))
        self.qty.setText(_translate("bill", "Quantity"))
        self.tot.setText(_translate("bill", "Total"))
        self.insert.setText(_translate("bill", "Insert"))
        self.update.setText(_translate("bill", "Update"))
        self.delete.setText(_translate("bill", "Delete"))
        self.select.setText(_translate("bill", "Select"))
        self.id.setText(_translate("bill", "id"))


if __name__ == "__main__":
    import sys
    app = QtWidgets.QApplication(sys.argv)
    bill = QtWidgets.QWidget()
    ui = Ui_bill()
    ui.setupUi(bill)
    bill.show()
    sys.exit(app.exec_())